Cargo.toml

```diff
@@ -11,13 +11,14 @@ rust-version = "1.60"
 authors = ["Christopher Berner <christopherberner@gmail.com>"]
 
 [lib]
-crate-type = ["cdylib", "rlib"]
```

Comment:
Is it necessary to remove this? I've forgotten why I had to add it, but I think it's for the Python support

Reply:
right, this should be reverted indeed.

Reply:
Although with `"cdylib"` it could not run tests on `metal` (which is kind of reasonable, as there could be no *system* libs). What if we just leave `crate-type = ["lib"]` there and let the compiler decide what to build? Can you test it with Python?

As far as I remember, last time I was trying to set `crate-type` programmatically about 2 years ago and the only way to do so was to run a bash script that changed Cargo.toml file depending on build conditions. Yet. requiring Cargo.toml edits before running baremetal tests sounds better than harming Python compatibility, thus it would be our worst case scenario option.

Reply:
can't reproduce python build by following readme instructions. Looks like current maturin version has different options organization and something else changed, but I've never tried that tool before to be sure.

Reply:
Hmm ok, may take me a bit to get to testing that, but I'll give it a go

Reply:
well, looking at the CI it already runs this on MacOS. So I guess it's fine :)